clients/rospy/src/rospy/rosconsole.py

```diff
@@ -135,13 +143,107 @@ def _rosconsole_cmd_get(argv):
     print(logger_level._current_levels[args[1]])
 
 
+class RosConsoleEcho(object):
+    LEVEL_STRING_NO_COLOR = {
+        Log.DEBUG: 'DEBUG',
+        Log.INFO : 'INFO ',
+        Log.WARN : 'WARN ',
+        Log.ERROR: 'ERROR',
+        Log.FATAL: 'FATAL',
+    }
+
+    LEVEL_STRING_COLOR = {
+        Log.DEBUG: '\033[92mDEBUG\033[0m',
+        Log.INFO : '\033[97mINFO \033[0m',
+        Log.WARN : '\033[93mWARN \033[0m',
+        Log.ERROR: '\033[91mERROR\033[0m',
+        Log.FATAL: '\033[95mFATAL\033[0m',
+    }
+
+    STRING_LEVEL = {
+        'debug': Log.DEBUG,
+        'info' : Log.INFO,
+        'warn' : Log.WARN,
+        'error': Log.ERROR,
+        'fatal': Log.FATAL,
+    }
+
+    def __init__(self, options):
+        self._level_string_map = self.LEVEL_STRING_NO_COLOR if options.nocolor else \
+                                 self.LEVEL_STRING_COLOR
+
+        self._filter = re.compile(options.filter)
+        self._level = self.STRING_LEVEL[options.level.lower()]
+        self._detail = options.detail
+
+        callback = self._once_callback if options.once else self._callback
+        rospy.Subscriber(options.topic, Log, callback)
+
+    def _print(self, msg):
+        print('[ {} ] [\033[1m{}\033[21m]: {}'.format(
+            self._level_string_map[msg.level], msg.name, msg.msg))
+
+        if self._detail:
+            stamp_sec = msg.header.stamp.to_sec()
+            stamp_tz = datetime.fromtimestamp(stamp_sec, tzlocal())
+
+            print('          [{} ({:.6f})] [{}]: {}:{}'.format(
+                stamp_tz, stamp_sec, msg.function, msg.file, msg.line))
+
+    def _callback(self, msg):
+        if self._filter.search(msg.name) and msg.level >= self._level:
+            self._print(msg)
+
+    def _once_callback(self, msg):
+        self._callback(msg)
+        rospy.signal_shutdown('Done')
+
+
+def _get_cmd_echo_argparse(prog):
+    parser = ArgumentParser(prog=prog, description='Print logger messages')
+
+    parser.add_argument('filter', metavar='FILTER', type=str, nargs='?', default='.*',
+                        help='regular expression to filter the logger name (default: %(default)s)')
+
+    parser.add_argument('level', metavar='LEVEL', type=str, nargs='?', default='warn',
+                        choices=RosConsoleEcho.STRING_LEVEL.keys(),
+                        help='minimum logger level to print (default: %(default)s)')
```

Comment:
The main reason I made them positional arguments is because I didn't know how to provide auto-complete hints with `rosbash` when I have something like `-l` (or `--level`).

I initially had `--filter` and `--level`. If I only have `--level`, do you if it's possible to have auto-complete hints in https://github.com/ros/ros/pull/168?

Reply:
I think it is possible. Manually implemented completion is usually a :zipper_mouth_face: In newer code I usually rely on `argcomplete` but that would be a drastic change.

I will leave it up to you if you want to change the positional arguments to options.

Reply:
I've had a quick look at bash completion and it seems it's definitely possible. I don't know how easy and clean it'd be though.

Since I really like the idea of having `--level`, so we can change the level only, I'd give it a try.
I'll come back to you soon, after updating both PRs.

Reply:
Changed to `-l` (`--level`) and update the rosbash auto-complete function to support it in https://github.com/ros/ros/pull/168